src/payment/strategies/sage-pay/sage-pay-payment-strategy.ts

```diff
@@ -22,18 +23,50 @@ export default class SagePayPaymentStrategy extends CreditCardPaymentStrategy {
     }
 
     execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment } = payload;
+        if (!payment) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        const { paymentData } = payment;
+
+        if (!paymentData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPayment);
+        }
+
+        if (this._isThreeDSTwoExperimentOn()) {
+            const newPaymentData = {
+                ...paymentData,
+                browser_info: getBrowserInfo(),
+            };
+
+            if ( payload.payment ) {
+                payload.payment.paymentData = newPaymentData;
+            }
+        }
+
         return super.execute(payload, options)
             .catch((error: Error) => {
                 if (!(error instanceof RequestError) || !some(error.body.errors, { code: 'three_d_secure_required' })) {
                     return Promise.reject(error);
                 }
 
                 return new Promise(() => {
-                    this._formPoster.postForm(error.body.three_ds_result.acs_url, {
-                        PaReq: error.body.three_ds_result.payer_auth_request,
-                        TermUrl: error.body.three_ds_result.callback_url,
-                        MD: error.body.three_ds_result.merchant_data,
-                    }, undefined, '_top');
+                    let payload: {};
```

Comment:
Do we need to declare a type here? You can leave it as `let payload;` and let typescript compiler infer it.

Reply:
sure thing!